File: backend/core/responses/baseAlbumResponse.py
from typing import Sequence

from pydantic import BaseModel

from backend.core.responses.baseArtistResponse import BaseArtistResponse
from backend.core.responses.baseSongResponse import BaseSongResponse


class BaseAlbumResponse(BaseModel):
    provider: str
    publicId: str
    name: str
    artists: Sequence[BaseArtistResponse]
    songs: Sequence[BaseSongResponse]
    releaseDate: str
    internalImageUrl: str

File: frontend/dto/baseAlbumResponse.ts
import { z } from "zod";
import { BaseArtistResponseSchema, BaseSongResponseSchema } from "@/dto";

export const BaseAlbumResponseSchema = z.object({
    provider: z.string(),
    publicId: z.string(),
    name: z.string(),
    artists: z.array(z.lazy(() => BaseArtistResponseSchema)),
    releaseDate: z.string(),
    internalImageUrl: z.string(),
    songs: z.array(z.lazy(() => BaseSongResponseSchema)),
});

export type BaseAlbumResponse = z.infer<typeof BaseAlbumResponseSchema>;
